### ISSUE TITLE
OAuth2 support

### ISSUE THREAD

**Issue by frol** [Yelp/bravado#193]
I see that there is a way to manage headers, so it is possible to set a token manually, but it would be nice to have OAuth2 support out of the box.

**Comment by wtfrank** [comment]
I agree - trapping exceptions to guard against auth token expiry every time you make a call feels like something that should be done internally.
